### DISCUSSION TITLE
Reference Resolution

### DISCUSSION THREAD

**Discussion by edzaffirm** [detekt/detekt#5533] [Q&A] [answered]
Hi! I'm attempting to write a custom rule. With this given code sample:
```
val foo = "foo"
some_call(foo)
``` 
I'd like to be able to resolve `foo` to its declaration then check that it's value `"foo"` is what I expect. Is this possible with Detekt?

I'm able to visit and grab the `KtReferenceExpression` for `foo`, but when I attempt to get its references, via `expression.references`, the list is empty, even though I'd expect it to refer to the variable declaration above. I'm playing around with the `bindingContext[BindingContext.REFERENCE_TARGET, ...]` which seems promising, but I'm unsure how to proceed from there.

**Comment by schalkms** [comment, marked as answer]
> I'd like to be able to resolve foo to its declaration then check that it's value "foo" is what I expect. Is this possible with Detekt?

This depends heavily on what you are trying to achieve. If you can statically determine the value of the variable, without actually running the program, then I think it’s possible.

I guess a good starting point is the **UnusedPrivateMember** rule of the detekt standard ruleset.
